src/e2e-test/java/io/cdap/cdap/ui/utils/Helper.java

```diff
@@ -87,4 +102,74 @@ public static String getNodeSelectorFromNodeIndentifier (NodeInfo node) {
       node.getNodeType() + "-" +
       node.getNodeId() + "\"]";
   }
+
+  public static void deployAndTestPipeline(String filename, String pipelineName) {
+    SeleniumDriver.openPage(Constants.PIPELINES_URL);
+    WaitHelper.waitForPageToLoad();
+    ElementHelper.clickOnElement(locateElementById("resource-center-btn"));
+    ElementHelper.clickOnElement(locateElementById("create-pipeline-link"));
+    if(!SeleniumDriver.getDriver().getCurrentUrl().contains("studio")) {
+      throw new RuntimeException("URL redirection for pipeline creation unsuccessful.");
+    }
+    WaitHelper.waitForPageToLoad();
+    WebElement upload_file = SeleniumDriver.getDriver()
+            .findElement(By.xpath("//*[@id='pipeline-import-config-link']/input[@type='file']"));
+
+    File pipelineJSONFile = new File("src/e2e-test/fixtures/" + filename);
+    String filePath = pipelineJSONFile.getAbsolutePath();
+    upload_file.sendKeys(filePath);
+
+//    ElementHelper.clickOnElement(
+//            WaitHelper.waitForElementToBeClickable(locateElementByCssSelector("div[class*='pipeline-name']"))
+//    );
+    // Below is the workaround as above(commented block) is not working
+    try {
+      WebElement pipelineNameContainer = locateElementByCssSelector("div[class*='pipeline-name']");
+      ElementHelper.clickOnElement(pipelineNameContainer);
+    } catch(org.openqa.selenium.StaleElementReferenceException ex) {
+      WebElement pipelineNameContainer = locateElementByCssSelector("div[class*='pipeline-name']");
+      ElementHelper.clickOnElement(pipelineNameContainer);
+    }
+
+    WebElement pipeline_name_input = locateElementById("pipeline-name-input");
+    pipeline_name_input.clear();
+    pipeline_name_input.sendKeys(pipelineName);
+    pipeline_name_input.sendKeys(Keys.RETURN);
+
+//    WebElement deployPipelineElm =  WaitHelper.waitForElementToBeClickable(
+//                    locateElementByCssSelector(getCssSelectorByDataTestId("deploy-pipeline")), 1000);
+//    deployPipelineElm.click();
+// Below is the workaround as above(commented block) is not working
+    WebElement deployPipelineElm = locateElementByCssSelector(getCssSelectorByDataTestId("deploy-pipeline"));
+    JavascriptExecutor jse = (JavascriptExecutor) SeleniumDriver.getDriver();
```

Comment:
Why do you need a Javascript executor for a click action?

Reply:
Changed with some helper functions and JS executor is not needed now.